ThreeBotPackages/tfgrid/workloads/actors/workload_manager.py

```diff
@@ -206,47 +208,61 @@ def _reservation_check(self, jsxobj):
+        if reservation.data_reservation._ddict != j.data.serializers.json.loads(payload):
+            reservation.next_action = "invalid"
+
+        if reservation.data_reservation.expiration_reservation < j.data.time.epoch:
+            reservation.next_action = "delete"
+            self._add_to_actionable_workload(reservation)
+
+        if reservation.next_action == "create":
+            if reservation.data_reservation.expiration_provisioning > j.data.time.epoch:
+                if reservation.customer_signature:
+                    if self._validate_customer_signature(reservation):
+                        reservation.next_action = "sign"
                     else:
-                        jsxobj.next_action = "invalid"
+                        reservation.next_action = "invalid"
             else:
-                jsxobj.next_action = "invalid"
+                reservation.next_action = "invalid"
 
-        if jsxobj.next_action == "sign":
-            signatures = jsxobj.signatures_provision
-            request = jsxobj.data_reservation.signing_request_provision
+        if reservation.next_action == "sign":
+            signatures = reservation.signatures_provision
+            request = reservation.data_reservation.signing_request_provision
             if self._request_check(payload, request, signatures):
-                jsxobj.next_action = "pay"
+                reservation.next_action = "pay"
 
-        if jsxobj.next_action == "pay":
-            if self._validate_farmers_signature(jsxobj):
-                jsxobj.next_action = "deploy"
+        if reservation.next_action == "pay":
+            if self._validate_farmers_signature(reservation):
+                reservation.next_action = "deploy"
+                self._add_to_actionable_workload(reservation)
 
-        if jsxobj.next_action == "deploy":
+        if reservation.next_action == "deploy":
             # Temporary change to simplify reservation flow for testing
-            # signatures = jsxobj.signatures_delete
-            # request = jsxobj.data_reservation.signing_request_delete
+            # signatures = reservation.signatures_delete
+            # request = reservation.data_reservation.signing_request_delete
             # if self._request_check(payload, request, signatures):
-            #     jsxobj.next_action = "delete"
+            #     reservation.next_action = "delete"
+            #     self._add_to_actionable_workload(reservation)
             pass
 
-        jsxobj.save()
-        return jsxobj
+        reservation.save()
+        return reservation
+
+    def _add_to_actionable_workload(self, reservation):
+        for _, w in iterate_over_workloads(reservation):
+            ra = self.workload_actionable_model.new()
+            ra.workload_id = gwid(reservation.id, w.workload_id)
+            ra.save()
+
+    def _remove_actionable_workload(self, workload_id):
+        ras = self.workload_actionable_model.find(workload_id=workload_id)
```

Comment:
There is no `.delete` method in bcdb?

Reply:
threre is a `delete` method but I'm not sure you can do `delete(workload_id=workload_id)`
@rkhamis  ?